src/ptc/util/misc.clj

```diff
@@ -122,3 +116,11 @@
   [& messages]
   (or (empty? messages)
       (apply = (map :properties messages))))
+
+(defn getenv!
+  "Get value of environment variable NAME or throw if nil."
+  [name]
+  (let [value (System/getenv name)]
+    (when (nil? value)
+      (throw (IllegalStateException. (str name " must not be nil"))))
+    value))
```

Comment:
Ends with a bang because it might throw?

Reply:
Yeah, I'm not exactly sure on the convention here... `?` for predicates makes sense and I though `!` was for side-effecty stuff (but everything in clojure can have effects)...